libs/ngx-mime/src/lib/core/alto-service/html.formatter.spec.ts

```diff
@@ -51,4 +69,51 @@ describe('HtmlFormatter', () => {
       '<p style="font-weight: bold">fakeString1<p/><p>fakeString2<p/>'
     );
   });
+
+  it('should mark match when hits are available', () => {
+    const hits: Hit[] = [{id: 1, index: 2, match: 'this', label: '', before: '', after: '', rects: []}]
+    const result = new HtmlFormatter(sanitizer, hits).altoToHtml(alto);
+
+    expect(result).toBe(
+      '<p><mark>this</mark> is a test.<p/>'
+    );
+  });
+
+  it('should mark multiple words', () => {
+    const hits: Hit[] = [{id: 1, index: 2, match: 'this', label: '', before: '', after: '', rects: []},
+    {id: 2, index: 3, match: 'test', label: '', before: '', after: '', rects: []},
+    {id: 3, index: 4, match: 'is', label: '', before: '', after: '', rects: []}]
+    const result = new HtmlFormatter(sanitizer, hits).altoToHtml(alto);
+
+    expect(result).toBe(
+      '<p><mark>this</mark> <mark>is</mark> a <mark>test</mark>.<p/>'
+    );
+  });
+
+  it('should mark single letter words', () => {
+    const hits: Hit[] = [{id: 1, index: 2, match: 'a', label: '', before: '', after: '', rects: []}]
+    const result = new HtmlFormatter(sanitizer, hits).altoToHtml(alto);
+
+    expect(result).toBe(
+      '<p>this is <mark>a</mark> test.<p/>'
+    );
+  });
+
+  it('should mark whole words and not partially', () => {
+    const hits: Hit[] = [{id: 1, index: 2, match: 'is', label: '', before: '', after: '', rects: []}]
+    const result = new HtmlFormatter(sanitizer, hits).altoToHtml(alto);
+
+    expect(result).toBe(
+      '<p>this <mark>is</mark> a test.<p/>'
+    );
```

Comment:
Tests should be structured like this:
Setup
<empty line>
Action
<empty line>
Expectation